st2common/st2common/models/api/keyvalue.py

```diff
@@ -114,17 +120,22 @@ def from_model(cls, model, mask_secrets=True):
         if 'id' in doc:
             del doc['id']
 
-        if model.expire_timestamp:
+        if getattr(model, 'expire_timestamp', None) and model.expire_timestamp:
             doc['expire_timestamp'] = isotime.format(model.expire_timestamp, offset=False)
 
         encrypted = False
-        if model.secret:
+        secret = getattr(model, 'secret', False)
+        if secret:
             encrypted = True
 
-        if not mask_secrets and model.secret:
+        if not mask_secrets and secret:
             doc['value'] = symmetric_decrypt(KeyValuePairAPI.crypto_key, model.value)
             encrypted = False
 
+        scope = getattr(model, 'scope', SYSTEM_KV_PREFIX)
```

Comment:
If we did design this from scratch we would have the `scope` property and assign it to `system`/`global` by default. The other choice would be to have `{{system.foo}}` and `{{system.user.foo}}` where `system` is still a prefix and what follows is possibly a scope i.e. `user` or something else in the future if we allow for user configurable scopes. I actually do prefer `system` transitioning to a scope although it is really from the scope perspective better named global. I wish I had been prescient and made space for this when I added ability to look up kv-store values in rules, actions etc.

 Taking this forward it would mean that when `{{system.foo}}` and `{{user.foo}}` we now grab 2 top level words i.e. `system` and `user` for the context addressable space. Not ideal but not really much worse that what we have already. I feel we should only grab `{{$user.foo}}` or some such obscure token where 2 purposes are served -
1. `$` suggest it will be replaced
2. `$user` is less-likely to appear in an addressable context